sklearnex/neighbors/common.py

```diff
@@ -264,3 +266,41 @@ def _onedal_gpu_supported(self, method_name, *data):
 
     def _onedal_cpu_supported(self, method_name, *data):
         return self._onedal_supported("cpu", method_name, *data)
+
+    def kneighbors_graph(self, X=None, n_neighbors=None, mode="connectivity"):
+        check_is_fitted(self)
+        if n_neighbors is None:
+            n_neighbors = self.n_neighbors
+
+        # check the input only in self.kneighbors
+
+        # construct CSR matrix representation of the k-NN graph
+        if mode == "connectivity":
+            A_ind = self.kneighbors(X, n_neighbors, return_distance=False)
+            xp, _ = get_namespace(A_ind)
+            n_queries = A_ind.shape[0]
+            A_data = xp.ones(n_queries * n_neighbors)
+
+        elif mode == "distance":
+            A_data, A_ind = self.kneighbors(X, n_neighbors, return_distance=True)
+            xp, _ = get_namespace(A_ind)
+            A_data = xp.reshape(A_data, (-1,))
+
+        else:
+            raise ValueError(
+                'Unsupported mode, must be one of "connectivity", '
+                f'or "distance" but got "{mode}" instead'
+            )
+
+        n_queries = A_ind.shape[0]
+        n_samples_fit = self.n_samples_fit_
+        n_nonzero = n_queries * n_neighbors
+        A_indptr = xp.arange(0, n_nonzero + 1, n_neighbors)
+
+        kneighbors_graph = sp.csr_matrix(
```

Comment:
I think it would be better to use `sp.csr_array` if possible.
Because scipy is in the process of switching to array interface. See the note here: https://docs.scipy.org/doc/scipy/reference/sparse.html

Reply:
Totally agree, unfortunately we are stuck with csr_matrix for conformance reasons to sklearn: https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/neighbors/_base.py#L1010